### PR TITLE
Add artifact & execution map to ListEvents endpoint

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Fixes https://github.com/kubeflow/metadata/issues/81

More filter options can be added later based on the artifact & execution properties.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/84)

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -472,7 +472,55 @@ func (s *Service) ListEvents(ctx context.Context, req *api.ListEventsRequest) (*
...
+		Artifacts:  artifacts,
+		Executions: executions,
 	}, err
 }
+
+func (s *Service) getArtifactsInEvents(events []*mlpb.Event) (map[int64]*mlpb.Artifact, error) {
+	results := make(map[int64]*mlpb.Artifact)
+	for _, e := range events {
+		id := e.GetArtifactId()
+		if _, exists := results[id]; exists {
```
Comment (neuromage) [inline service/service.go]: nit: `exists` => `ok` as that's the common way to write this ('comma ok idiom'). Feel free to ignore :-)

Comment (zhenghuiwang) [reply service/service.go]: I will use this idiom in future PRs :-) Thanks

**Comment by zhenghuiwang** [comment]
Thanks for the review!

/approve

**Pull request merged**